src/ios/GeolocationPlugin.m

```diff
@@ -0,0 +1,314 @@
+    CDVPluginResult* pluginResult = nil;
+    coord = [command.arguments objectAtIndex:0];
+    
+    
+    //[self InitData];
+    
+    if (![self.thread isExecuting]) {
+        NSLog(@"Starting thread...");
+        loopflag = TRUE;
+        [self.thread start];
+    }
+}
+
+-(NSThread*)thread{
+    if (!_thread) {
+        _thread = [[NSThread alloc]initWithTarget:self selector:@selector(longloop) object:nil];
+    }
+    return _thread;
+}
+-(void)longloop{
+    while (loopflag) {
+        [self performSelector:@selector(InitData) onThread:[NSThread mainThread] withObject:[NSNumber numberWithLong:3] waitUntilDone:YES];
+        sleep(75);
+    }
+}
+
+
+-(void)stop:(CDVInvokedUrlCommand *)command
+{
+    loopflag = FALSE;
+    if ([self.thread isExecuting]) {
+        NSLog(@"Stopping thread...");
+        [self.thread cancel];
+        
+        self.thread = nil;
+    }
+}
+
+-(void)InitData{
+    [self CreateDataBase];
+    EnterRegion = @"NO";
+    // Add location Manager class
+    if (nil == self.locationManager) {
+        self.locationManager = [[CLLocationManager alloc] init];
+        // Get version of iOS
+        if ([[[UIDevice currentDevice] systemVersion] floatValue] >= 8.0) {
+            
+            [ self.locationManager requestWhenInUseAuthorization];
+            
+            [self.locationManager requestAlwaysAuthorization];
+        }
+    }
+    now     = [NSDate date];
+    now1    = [NSDate date];
+    
+    if (nil == self.locationManager)
+        self.locationManager = [[CLLocationManager alloc] init];
+    self.locationManager.delegate = self;
+    
+    
+    [self.locationManager startMonitoringSignificantLocationChanges];
+    [self.locationManager startUpdatingLocation];
+    [self regionStart];
+}
+
+
+// recieve GPS updated location
+- (void)locationManager:(CLLocationManager *)manager didUpdateLocations:(NSArray *)locations {
+    
+    
+    location = [locations lastObject];
+    
+    NSDate* timestamp = location.timestamp;
+    
+    
+    
+    NSTimeInterval age = [NSDate.date timeIntervalSinceDate:location.timestamp];
+    
+    if (fabs([now timeIntervalSinceDate:timestamp]) < 60){
+        
+        NSLog(@"Time 60 second print");
+        
+        if (fabs([now1 timeIntervalSinceDate:timestamp]) > 15) {
+            
+            now1 = timestamp;
+
+            
+            DateFormatter = [[NSDateFormatter alloc] init];
+            [DateFormatter setDateFormat:@"yyyyMMdd'T'HH:mm:ss"];
+            [self InsertData];
+        }
+        
+    }
+    else{
+        NSLog(@"Time 60 more than print");
+        [self MakeJSON_Make ];
+        [self stopUpdatingLocationWithMessage:NSLocalizedString(@"Error", @"Error")];
+        
+    }
+}
+
+
+
+- (void)stopUpdatingLocationWithMessage:(NSString *)state {
+    
+    [self.locationManager stopUpdatingLocation];
+    self.locationManager.delegate = nil;
+}
+
+-(void)regionStart{
+    NSArray *ns_coord = [coord componentsSeparatedByString:@","];
```

Comment:
`coord` should be passed as a parameter here, not as global variable. 

Reasons: 

1) if javascript calls this twice for some reason we can have inconsistent global data and the plugin might even crash the app.
2) when passing coord as a parameter the method can be re-used for multiple coordinates not just once